### PR TITLE
Settings_Profile_UI creation

### PR DESCRIPTION
**Pull request by DavidSurkov**
image

Profile (edit) page has been finished. Please check if some fixes needed.
Did not use react-hook-form. Because it was a complex form with different layers.
Custom validation has been applied.

### CODE REVIEW

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      });
+      return newState;
+    });
+  };
+
+  const [skillsOptions, setSkillsOptions] = useState([
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:76]: no, 1) move your moks to some mock constant after imports, 2d) data from backend will come without - true and false. I recomend set to state only selected items, and rest will come from backend. if you want go with way, anyway you will need map array that come from backend somewhere in useEffect to this new format, and then send to backend only selected items, which mean again transformation

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        if (index === i) {
+          return { ...e, value: !e.value };
+        }
+        return e;
+      });
+    });
+  };
+
+  const [englishOption, setEnglishOption] = useState('Pre-intermediate');
+  const englishOptions = [
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:115]: move above the hooks, dont save constants in your hooks or components

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    { label: 'Pre-intermediate', value: 'Pre-intermediate' },
+    { label: 'Intermediate', value: 'Intermediate' },
+    { label: 'Upper-intermediate', value: 'Upper-intermediate' },
+  ];
+  const onEnglishOptionChange = ({ target: { value } }: RadioChangeEvent) => {
+    setEnglishOption(value);
+  };
+
+  const [position, setPosition] = useState('');
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:124]: types

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  const [price, setPrice] = useState<number>(0);
+  const onPriceChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (+e.currentTarget.value < 0) {
+      window.alert('Wrong price');
+    } else {
+      setPrice(+e.currentTarget.value);
+    }
+  };
+
+  const [description, setDescription] = useState('');
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:149]: types
move all use state together to the top, then useEffects and after onClick and other functions

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+`;
+const SkillButton = styled.button<{ selected: boolean }>`
+  width: 200px;
+  border-radius: 10px;
+  margin: 5px;
+  height: 30px;
+  background-color: ${props => (props.selected ? 'dodgerblue' : '')};
+  color: ${props => (props.selected ? 'white' : '')};
+`;
+const EnglishLevel = styled.div``;
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:54]: Was thinking for future changes

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 6]
```diff
@@ -0,0 +1,214 @@
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { Radio } from 'antd';
+import { TextareaWithDates } from './components/TextareaWithDates';
+import { useOnDataChange } from './hooks/use-on-data-change';
+import { Image, Alert } from 'antd';
+
+//Styles
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:8]: done

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+const SkillsButtonsBlock = styled.div`
+  max-width: 840px;
+`;
+const SkillButton = styled.button<{ selected: boolean }>`
+  width: 200px;
+  border-radius: 10px;
+  margin: 5px;
+  height: 30px;
+  background-color: ${props => (props.selected ? 'dodgerblue' : '')};
+  color: ${props => (props.selected ? 'white' : '')};
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:52]: done

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+  width: 500px;
+  height: ${props => props.height};
+  resize: none;
+`;
+
+const SaveButton = styled.button`
+  width: 140px;
+  height: 50px;
+  border-radius: 10px;
+  background-color: #5db501;
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:66]: done

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+  return (
+    <Container>
+      <Block>
+        <ProfilePhoto>
+          <span>Profile photo</span>
+          <Image
+            width={200}
+            src={
+              file64?.toString() ||
+              'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png'
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:120]: moved to the constant

File: src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  const [text, setText] = useState<string>(props.item.info || '');
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+
+  useEffect(() => {
+    let error = props.item.error;
+    if (!text && isTouched) {
+      error = true;
+    } else {
+      error = false;
+    }
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx:52]: done

File: src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+export const TextareaWithDates = (props: ITextareaWithDates) => {
+  const [startDate, setStartDate] = useState<string>(
+    props.item.dateStart || new Date().toISOString(),
+  );
+  const [endDate, setEndDate] = useState<string>(props.item.dateEnd || new Date().toISOString());
+  const [text, setText] = useState<string>(props.item.info || '');
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+
+  useEffect(() => {
+    let error = props.item.error;
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx:47]: done

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts (hunk 3 above)
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:124]: done

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts (hunk 4 above)
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:149]: as long as state is not reusable in many places it makes more sense to keep them next to the logic. easier to read

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+          <Price>
+            <h2>Price</h2>
+            <input onChange={onPriceChange} value={price} type="number" />
+            <span>$</span>
+          </Price>
+        </div>
+      </Block>
+      <Skills>
+        <h2>Skills</h2>
+        <span>You must mark at least 3 skills</span>
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:151]: done

File: src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+      index: props.index,
+      item: { info: text, dateStart: startDate, dateEnd: endDate, error: error },
+      key: props.objectKey,
+    });
+    if (!isTouched) setIsTouched(true);
+  }, [startDate, endDate, text]);
+
+  const onChangeStartDate = (e: ChangeEvent<HTMLInputElement>) => {
+    if (new Date(e.currentTarget.value) > new Date(endDate)) {
+      window.alert('Wrong Date');
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/components/TextareaWithDates.tsx:63]: done

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts (hunk 2 above)
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:115]: done

File: src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts (hunk 1 above)
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts:76]: 1) moved;
2) when I see the back api, I'll adapt my state

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 14]
```diff
@@ -4,10 +4,19 @@ import { Radio } from 'antd';
 import { TextareaWithDates } from './components/TextareaWithDates';
 import { useOnDataChange } from './hooks/use-on-data-change';
 import { Image, Alert } from 'antd';
+import { useTranslation } from 'react-i18next';
+
+const colors = {
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:9]: нет, кгнстанты должны быть гдобальные для всего проекта
что бы везде балый был 00000 а не в одном месте 000001 а в другом 000002, тоже самое для фото, скажут поменять, ты на второй день забудешь к ваком компоненте эта константа, проще найти константу в файлике с константами, чем все компоненты перерывать,

Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:9]: done. thanks

File: src/pages/settings-page/SettingsPage.tsx [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+  margin: 0 auto;
+  max-width: 1000px;
+`;
+
+const Link = styled(NavLink)`
+  font-size: 24px;
+  margin-right: 5px;
+  text-decoration: none;
+  padding: 0 5px;
+  color: black;
```
Comment (lyudad) [inline src/pages/settings-page/SettingsPage.tsx:15]: constant

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+import { Image, Alert } from 'antd';
+import { useTranslation } from 'react-i18next';
+import { defaultProfilePhoto } from 'constants/links';
+import {
+  BLACK_COLOR,
+  DEFAULT_COLOR,
+  DODGERBLUE_COLOR,
+  GREEN_COLOR,
+  WHITE_COLOR,
+} from '../../../../constants/colors';
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:15]: ?

File: src/pages/settings-page/SettingsPage.tsx (hunk 15 above)
Comment (DavidSurkov) [reply src/pages/settings-page/SettingsPage.tsx:15]: noted and fixed

File: src/pages/settings-page/SettingsPage.tsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+  &.active {
+    border-width: 1px 1px 0 1px;
+    border-color: black;
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/SettingsPage.tsx:18]: noted and fixed

File: src/pages/settings-page/components/ContactInfo/ContactInfo.tsx [hunk 18]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const Container = styled.div`
+  border: 1px solid black;
```
[the remark this answers is not in the document]
Comment (DavidSurkov) [reply src/pages/settings-page/components/ContactInfo/ContactInfo.tsx:5]: noted and fixed

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx (hunk 16 above)
Comment (DavidSurkov) [reply src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:15]: noted and fixed

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+    ));
+  }, [skillsOptions]);
+
+  return (
+    <Container>
+      <Block>
+        <ProfilePhoto>
+          <Title>{`${t('ProfileEdit.profilePhotoTitle')}`}</Title>
+          <Image width={200} src={file64?.toString() || defaultProfilePhoto} />
+          <input type={'file'} onChange={onChangePhotoHandler} />
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:129]: why you dint use react hook forms controller and input?

File: src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+                message={`${t('ProfileEdit.positionAlert')}`}
+                type="warning"
+                showIcon
+                closable
+              />
+            )}
+          </Position>
+          <Category>
+            <Title>{`${t('ProfileEdit.categoryTitle')}`}</Title>
+            <select onChange={onCategoryChange} value={category}>
```
Comment (lyudad) [inline src/pages/settings-page/components/ProfileEdit/ProfileEdit.tsx:147]: same, you need to handler each field without form library

**Pull request merged**